app/views/ProfileView/index.tsx

```diff
@@ -334,39 +271,13 @@ class ProfileView extends React.Component<IProfileViewProps, IProfileViewState>
 			EventEmitter.emit(LISTENER, { message: I18n.t('Avatar_changed_successfully') });
 			this.init();
 		} catch (e) {
-			this.handleError(e, 'resetAvatar', 'changing_avatar');
-		}
-	};
-
-	pickImage = async () => {
-		const { Accounts_AllowUserAvatarChange } = this.props;
-
-		if (!Accounts_AllowUserAvatarChange) {
-			return;
-		}
-
-		const options = {
-			cropping: true,
-			compressImageQuality: 0.8,
-			freeStyleCropEnabled: true,
-			cropperAvoidEmptySpaceAroundImage: false,
-			cropperChooseText: I18n.t('Choose'),
-			cropperCancelText: I18n.t('Cancel'),
-			includeBase64: true
-		};
-		try {
-			logEvent(events.PROFILE_PICK_AVATAR);
-			const response: Image = await ImagePicker.openPicker(options);
-			this.setAvatar({ url: response.path, data: `data:image/jpeg;base64,${response.data}`, service: 'upload' });
-		} catch (error) {
-			logEvent(events.PROFILE_PICK_AVATAR_F);
-			console.warn(error);
+			handleError(e, 'resetAvatar', 'changing_avatar');
 		}
 	};
 
-	pickImageWithURL = (avatarUrl: string) => {
-		logEvent(events.PROFILE_PICK_AVATAR_WITH_URL);
-		this.setAvatar({ url: avatarUrl, data: avatarUrl, service: 'url' });
+	handleEditAvatar = () => {
+		const { navigation } = this.props;
+		navigation.navigate('ChangeAvatarView', { fromUser: true });
```

Comment:
We talked about this `fromUser` prop.
Perhaps you could make it more dynamic while increasing semantic?

Think like this:
`ChangeAvatarView` is a screen that behaves differently based on the context, because it can be called from `ProfileView`, `RoomInfoView` and `RoomInfoEditView`.
You created `fromUser` to make it behave differently for `ProfileView`.
What if you create TS types to specify the context?

```
type TChangeAvatarViewContext = 'profile' | 'room';
```

Add it to the screen's props and then TS is going to autocomplete it for you.